src/flint/fmpq.jl

```diff
@@ -883,6 +883,15 @@ end
 
 (a::FlintRationalField)(b::fmpq) = b
 
+function (a::FlintIntegerRing)(b::fmpq)
+   z = fmpz()
+   ccall((:fmpq_denominator, libflint), Nothing, (Ref{fmpz}, Ref{fmpq}), z, b)
+   isone(denominator(b)) || error("Denominator must be 1")
```

Comment:
Is there a flint function to do this allocation free?

Reply:
oops. I royally messed that line up. I was trying to make zero garbage.